### PR TITLE
Label property outlier

### LINKED ISSUE #756: [FEAT][CV]Label outlier check
The goal of this check is to detect outlier classes, according for some label properties. 

What is an outlier class? A class who's average properties (abundance for classification, avg bbox size or count in detection) are significantly different compared to other classes.

Writing this check requires writing an algorithm to detect outlier classes in 1 dimensional properties. Example for such an algorithm - same algorithm used in [boxplots](https://www.simplypsychology.org/boxplots.html#:~:text=An%20outlier%20is%20an%20observation,whiskers%20of%20the%20box%20plot.)

### PR DESCRIPTION
**Pull request by matanper**
resolve #756

### CODE REVIEW

File: deepchecks/vision/checks/distribution/image_property_outliers.py [hunk 1]
```diff
@@ -9,36 +9,26 @@
...
 
-from deepchecks.vision.utils.image_functions import prepare_thumbnail
-
 
-class ImagePropertyOutliers(SingleDatasetCheck):
-    """Find outliers images with respect to the given image properties.
+class ImagePropertyOutliers(AbstractPropertyOutliers):
+    """Find outliers images with respect to the given properties.
 
-    The check computes several image properties and then computes the number of outliers for each property.
```
Comment (noamzbr) [inline deepchecks/vision/checks/distribution/image_property_outliers.py:35]: why remove the word "image" here? It is image properties.

File: deepchecks/vision/checks/distribution/label_property_outliers.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+from deepchecks.vision.utils.image_functions import draw_bboxes
+from deepchecks.vision.vision_data import TaskType, VisionData
+
+__all__ = ['LabelPropertyOutliers']
+
+
+class LabelPropertyOutliers(AbstractPropertyOutliers):
+    """Find outliers labels with respect to the given properties.
+
+    The check computes several properties and then computes the number of outliers for each property.
```
Comment (noamzbr) [inline deepchecks/vision/checks/distribution/label_property_outliers.py:27]:
```suggestion
    The check computes several label properties and then computes the number of outliers for each property.
```

File: deepchecks/vision/checks/distribution/label_property_outliers.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        The scale to multiply the IQR range for the outliers detection
+    """
+
+    def get_default_properties(self, data: VisionData):
+        """Return default properties to run in the check."""
+        if data.task_type == TaskType.CLASSIFICATION:
+            raise DeepchecksProcessError('task type classification does not have default label '
+                                         'properties for label outliers.')
+        elif data.task_type == TaskType.OBJECT_DETECTION:
+            return label_prediction_properties.DEFAULT_OBJECT_DETECTION_LABEL_PROPERTIES
```
Comment (noamzbr) [inline deepchecks/vision/checks/distribution/label_property_outliers.py:51]: where is the part that filters out the class_id properties?

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+Label Property Outliers
+=======================
+This notebooks provides an overview for using and understanding the image property
```
Comment (noamzbr) [inline docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:5]: update this

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+outliers check, used to detect outliers in simple image properties in a dataset.
+
+**Structure:**
+
+* `Why Check for Outliers? <#why-check-for-outliers>`__
```
Comment (noamzbr) [inline docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:10]:
```suggestion
* `Why Check for Label Outliers? <#why-check-for-outliers>`__
```

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+train_data = load_dataset(train=True, object_type='VisionData')
+check = LabelPropertyOutliers()
+result = check.run(train_data)
+result
+
+#%%
+# Observe Graphic Result
+# ^^^^^^^^^^^^^^^^^^^^^^
+# The check shows a section for each property. In each section we show the number of outliers and the non-outlier
```
Comment (noamzbr) [inline docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:59]:
```suggestion
# The check displays a section for each property. In each section we show the number of outliers and the non-outlier
```

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+# property range, and also the images with the lowest and highest values for the property. In addition, if the property
+# returns a value per bounding box, then we show only the relevant bounding box which resulted in the outlier result.
```
Comment (noamzbr) [inline docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:61]:
```suggestion
# returns a value per bounding box, we then show only the relevant bounding boxes which resulted in the outlier result.
```

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+#
+# For example in property "Bounding Box Area (in pixels)" we can see that 80 outliers were found. Now we can inspect
+# the samples and decide if we wish to ignore these kinds of samples or if we would
+# like the model to be able to support them, in which case we may take a close look into the model's predictions on
+# these samples.
+#
+# Observe Result Value
+# ^^^^^^^^^^^^^^^^^^^^
+# The check returns CheckResult object with a property 'value' on it which contain the information that was calculated
```
Comment (noamzbr) [inline docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:70]: What information?

File: tests/vision/checks/distribution/image_property_outliers_test.py [hunk 9]
```diff
@@ -66,7 +66,7 @@ def test_image_property_outliers_check_mnist(mnist_dataset_train, device):
     assert_that(result.value, has_entries({
         'Brightness': has_entries({
             'indices': has_length(610),
-            'lower_limit': close_to(2.936, .001),
+            'lower_limit': close_to(6.487, .001),
```
Comment (JKL98ISR) [inline tests/vision/checks/distribution/image_property_outliers_test.py:69]: why has this changed?

Comment (matanper) [reply tests/vision/checks/distribution/image_property_outliers_test.py:69]: I changed some things in the algo

File: deepchecks/vision/checks/distribution/image_property_outliers.py (hunk 1 above)
Comment (matanper) [reply deepchecks/vision/checks/distribution/image_property_outliers.py:35]: it's kinda redundant from the name, don't you think? anyway added back

File: deepchecks/vision/checks/distribution/label_property_outliers.py (hunk 3 above)
Comment (matanper) [reply deepchecks/vision/checks/distribution/label_property_outliers.py:51]: in the abstract

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py (hunk 7 above)
Comment (matanper) [reply docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:61]: that's not correct I indeed meant we show a single bounding box, not boxes

File: docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py (hunk 8 above)
Comment (matanper) [reply docs/source/examples/vision/checks/distribution/source/plot_label_property_outliers.py:70]: It's kinda self-explanatory, seems unnecessary to also print the result and also explain about it
```
{ 'Bounding Box Area' : {
    'indices': [1,2,3,100]
    'lower_limit': 18,
    'upper_limit': 12000
}}
```

**Review by ItayGabbay — APPROVED** [review]
Approving the tests and docs sections

**Review by JKL98ISR — APPROVED** [review]

**Pull request merged**